python-package/xgboost/core.py

```diff
@@ -1001,6 +1001,69 @@ def feature_types(self, feature_types):
         self._feature_types = feature_types
 
 
+class DeviceQuantileDMatrix(DMatrix):
+    """Device memory Data Matrix used in XGBoost.
+
+    DMatrix is a internal data structure that used by XGBoost
```

Comment:
Do we need to mention `gpu_hist` explicitly ?

Reply:
Done, added a little more documentation.